### PR TITLE
Default to array

### LINKED ISSUE #22: Default to array instead of object if all update keys are numbers
Extracted from https://github.com/substantial/updeep/issues/20

```js
u.({ 0: 'hi' }, null) // => ['hi']
```

### PR DESCRIPTION
**Pull request by astephenb**
fixes #22

### CODE REVIEW

File: test/updateIn-spec.js [hunk 1]
```diff
@@ -27,6 +27,12 @@ describe('u.updateIn', () => {
     expect(result).to.eql({ a: [0, 3, 0] });
   });
 
+  it('can create array if all keys are numbers', () => {
+    const result = u.updateIn('a.1', 3, null);
+
+    expect(result).to.eql({ a: [, 3] });
```
Comment (astephenb) [inline test/updateIn-spec.js]: This leaves the linter unhappy, but I am not sure how to test this. Creating a new array and setting a value at an index `i` where `i` is greater than array.length results in some an array with some elements missing [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#Relationship_between_length_and_numerical_properties). This seems fine from an application level, but testing it in this way makes the linter unhappy. 

Should I just disable the linter warnings for this line?

File: lib/util/newDefaultObject.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import isEmpty from './isEmpty';
+
+function isArrayIndex(n) {
+  return /^[0-9]+$/.test(n);
```
Comment (aaronjensen) [inline lib/util/newDefaultObject.js]: let's use this one: 

```js
function isInt(value) {
  if (isNaN(value)) {
    return false;
  }
  var x = parseFloat(value);
  return (x | 0) === x;
}
```

via http://stackoverflow.com/questions/14636536/how-to-check-if-a-variable-is-an-integer-in-javascript

File: test/updateIn-spec.js (hunk 1 above)
Comment (aaronjensen) [inline test/updateIn-spec.js]: does `[undefined, 3]` pass?

File: lib/util/newDefaultObject.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+function newDefaultObject(object, updates) {
+  if (typeof object === 'undefined' || object === null) {
+    return arrayOrObject(updates);
+  }
+
+  return object;
+}
+
+export default newDefaultObject;
```
Comment (aaronjensen) [inline lib/util/newDefaultObject.js]: I might just call this `defaultObject`... it doesn't always new something up, but it does always "default" it.

**Comment by aaronjensen** [comment]
This looks, good, once you get stuff passing can you rebase into a single commit and update the readme/changelog too? Thanks!

File: test/updateIn-spec.js (hunk 1 above)
Comment (astephenb) [inline test/updateIn-spec.js]: It doesn't.

File: test/updateIn-spec.js [hunk 4]
```diff
@@ -27,6 +27,12 @@ describe('u.updateIn', () => {
     expect(result).to.eql({ a: [0, 3, 0] });
   });
 
+  it('can create array if all keys are numbers', () => {
+    const result = u.updateIn('a.1', 3, null);
+
+    expect(result).to.eql({ a: [, 3] }); // eslint-disable-line no-sparse-arrays
```
Comment (astephenb) [inline test/updateIn-spec.js]: I'm actually surprised that this is valid syntax. It's not pretty though. Another option would be:
```javascript
expect(result.a[1]).to.eql(3);
```

Comment (aaronjensen) [inline test/updateIn-spec.js]: well, you'd also need to test that it is an array if you did that. You could also just use the 0 index. I'm good w/ any of these

Comment (astephenb) [inline test/updateIn-spec.js]: I'll do 0 index

**Pull request merged**